test/Test/Stan/Analysis/AntiPattern.hs

```diff
@@ -130,3 +131,39 @@ unsafeFunctionsSpec analysis = describe "STAN-0212: Unsafe functions" $ do
         checkObservation AntiPattern.stan0212 22 30 52
     it "Find: unsafeFixIO" $
         checkObservation AntiPattern.stan0212 25 19 30
+
+compareSpec :: Analysis -> Spec
+compareSpec analysis = describe "STAN-0214: Replace multiple comparison operators" $ do
+    let checkObservation = observationAssertMulti
+            "Target/AntiPattern/Stan0214.hs"
+            "Target.AntiPattern.Stan0214"
+            analysis
+            AntiPattern.stan0214
+    let noObservation = noObservationAssert
+            "Target/AntiPattern/Stan0214.hs"
+            "Target.AntiPattern.Stan0214"
+            analysis
+            AntiPattern.stan0214
+
+    it "Finds: < and >" $
+        checkObservation 7 1 10 23
+    it "Finds: == and <" $
+        checkObservation 13 1 16 23
+    it "No warning on: == on different expressions" $ do
+        noObservation 19
+        noObservation 20
+        noObservation 21
+        noObservation 22
+    it "No warning on: Single >=" $ do
+        noObservation 25
+        noObservation 26
+        noObservation 27
+    it "No warning on: >= with different constants" $ do
+        noObservation 30
+        noObservation 31
+        noObservation 32
+        noObservation 33
+        noObservation 34
+        noObservation 35
+    xit "Handles functions with pattern-matching" $
```

Comment:
:+1: can figure out this one later! This is already huge and amazing :tangerine: